commons/src/main/java/org/infinispan/commons/dataconversion/MediaType.java

```diff
@@ -91,13 +92,20 @@
    private static final double DEFAULT_WEIGHT = 1.0;
    private static final Charset DEFAULT_CHARSET = UTF_8;
 
-   private Map<String, String> params = new HashMap<>(2);
-   private String type;
-   private String subType;
-   private String typeSubtype;
-   private transient double weight;
+   private final Map<String, String> params = new HashMap<>(2);
+   private final String type;
+   private final String subType;
+   private final String typeSubtype;
+   private final transient double weight;
 
-   MediaType() {}
+   @ProtoFactory
+   MediaType(String tree) {
```

Comment:
This looks a bit odd. Why not add ```@ProtoFactory``` directly to ```parse``` method.

Reply:
Good call! For some reason I didn't think `@ProtoFactory` would work on non-constructors